lang/machine.md

```diff
@@ -43,6 +43,44 @@ struct StackFrame<M: Memory> {
 }
 ```
 
+Next, we define the functions necessary to create and run a machine.
+
+```rust
+impl<M: Memory> Machine<M> {
+    pub fn new(prog: Program) -> Machine<M> {
+        let start_fn = prog.functions[prog.start];
+
+        let null_ptr = Pointer {
+            addr: Int::ZERO,
+            provenance: None
+        };
+
+        // setup the initial stack frame.
+        let init_frame = StackFrame {
+            func: start_fn,
+            locals: Map::new(),
+            // The initial function has no `caller_ret_place`.
+            caller_ret_place: null_ptr,
```

Comment:
```suggestion
            // The initial function has no caller and hence no `caller_ret_place`.
            caller_ret_place: null_ptr,
```

Reply:
Yep, that sounds better. It's squashed in!